examples/pyolite - images.ipynb

```diff
@@ -0,0 +1,8 @@
+{
+  "metadata": {
+    "orig_nbformat": 4
+  },
+  "nbformat_minor": 4,
+  "nbformat": 4,
+  "cells": []
```

Comment:
Looks like the notebook is empty?

Reply:
Indeed... looks like a bug in jupyterlite that when I save the notebook and download it, it is empty.

Reply:
Manually recreated the notebook, as there doesn't seem a way to make a new notebook (with contents) from jupyterlite.

Reply:
Ah looks like this might be a bug then. Would you like to open a new issue to track this?

Normally the download command from the command palette should work:

![image](https://user-images.githubusercontent.com/591645/124792920-c5dcf480-df4d-11eb-946f-0cb049e19ce5.png)

Reply:
Might be related: https://github.com/jtpio/jupyterlite/issues/229

Reply:
There is an issue in opening a file in the notebook. I'll take a look at that. Perhaps there is a standard way of opening files in jupyterlite?

Reply:
By opening a file, do you mean something like in https://github.com/jtpio/jupyterlite/discussions/91?

Reply:
From the notebook:

![Screenshot from 2021-07-07 11-03-43](https://user-images.githubusercontent.com/168568/124807809-0b300080-df13-11eb-8009-01e67811f5e1.png)

Reply:
BTW, PIL is already imported, so this additional patch is not adding any additional time to the current load time. PIL is import via matplotlib. So this is going to be great to have a better system!

Re: moving to a lazy import-patching system, there are at least two options:

1. write the bare minimum necessary ourselves for our minimal needs
2. use a package like [importhook](https://github.com/brettlangdon/importhook) that appears to work, is tested, and maintained

What would you like to pursue?

Reply:
importhook looks nice and has no dependencies. a perusal of some of the crazy stuff needed in the example notebooks shows some of the challenges, where to get to something functioning in `sys.modules`, we need to (for now) to install particular versions of upstreams.

Looking at those, we might need something that lets us codify `preimport` (and `preinstall`) hooks. However, as we don't have persistence in the WASM filesystem, these are the same... but we're working on that upstream.

The `preinstall` might be solveable through a patch of `micropip` (which _could_ be done via `importhook`). 

I feel like the long con would be for site builders to be able to add required patches via config file, as their end users shouldn't even know they're doing anything special (aside from _perhaps_ a context manager or a magic).

Reply:
> BTW, PIL is already imported, so this additional patch is not adding any additional time to the current load time. PIL is import via matplotlib. So this is going to be great to have a better system!

Thanks @dsblank

So we can probably go ahead with this patch and try to improve things in follow-up issues and PRs?

Reply:
> So we can probably go ahead with this patch and try to improve things in follow-up issues and PRs?

Sounds like a plan. I updated the notebook to save and then load, and that appears to work well. So, I think this PR is complete. 

I'll open up another issue so we can discuss the patching issue. 

Thanks all!